galsim/sed.py

```diff
@@ -213,77 +277,210 @@ def __call__(self, wave):
             if wmax > self.red_limit + extrapolation_slop:
                 raise ValueError("Requested wavelength ({0}) is redder than red_limit ({1})"
                                  .format(wmax, self.red_limit))
-        wave_factor = 1.0 + self.redshift
-        # figure out what we received, and return the same thing
-        # option 1: a numpy array
-        if isinstance(wave, np.ndarray):
-            return self._rest_photons(wave / wave_factor)
-        # option 2: a tuple
-        elif isinstance(wave, tuple):
-            return tuple(self._rest_photons(np.array(wave) / wave_factor))
-        # option 3: a list
+
+    def _call_fast(self, wave):
+        """ Return either flux in photons / sec / cm^2 / nm, or dimensionless normalization.
+
+        Assumes that self._spec has already been transformed to accept correct wavelength units and
+        yield correct flux units.
+
+        @param wave  Wavelength in nanometers.
+        @returns     Flux or normalization.
+        """
+        self._check_bounds(wave)
+
+        # Create a fast version of self._spec by constructing a LookupTable on self.wave_list
+        if not hasattr(self, '_fast_spec'):
+            if (self.wave_type == units.nm
+                and self.flux_type == _photons):
+                    self._fast_spec = self._spec
+            else:
+                if len(self.wave_list) == 0:
+                    if self.spectral:
+                        self._fast_spec = self._rest_nm_to_photons
+                    else:
+                        self._fast_spec = self._rest_nm_to_dimensionless
+                else:
+                    x = self.wave_list / (1.0 + self.redshift)
+                    if self.spectral:
+                        f = self._rest_nm_to_photons(x)
+                    else:
+                        f = self._rest_nm_to_dimensionless(x)
+                    self._fast_spec = galsim.LookupTable(x, f, interpolant='linear')
+
+        if isinstance(wave, tuple):
+            return tuple(self._fast_spec(np.array(wave) / (1.0 + self.redshift)))
         elif isinstance(wave, list):
-            return list(self._rest_photons(np.array(wave) / wave_factor))
-        # option 4: a single value
+            return list(self._fast_spec(np.array(wave) / (1.0 + self.redshift)))
+        else:  # ndarray or scalar
+            return self._fast_spec(wave / (1.0 + self.redshift))
+
+    def _call_slow(self, wave):
+        """ Return flux in photons / sec / cm^2 / nm or dimensionless normalization.
+
+        Uses self._spec that has not been pre-transformed for desired units, instead does all unit
+        conversions inside this method.
+
+        @param wave  Wavelength.  If not a Quantity, then assumed units are nanometers.
```

Comment:
I recommend that when referring to classes that live in astropy, indicate that explicitly.